File: server/dotnet/sdk-dotnet-example/Business/Domain/Payments/PaymentDetails/DccProposal.cs
namespace Business.Domain.Payments.PaymentDetails;

public class DccProposal
{
    public AmountOfMoney? BaseAmount { get; set; }

    public string? DisclaimerDisplay { get; set; }

    public string? DisclaimerReceipt { get; set; }

    public RateDetails? Rate { get; set; }

    public AmountOfMoney? TargetAmount { get; set; }
}

File: server/python/app/application/domain/payments/payment_details/dcc_proposal.py
from dataclasses import dataclass
from typing import Optional

from app.application.domain.payments.payment_details.amount_of_money import AmountOfMoney
from app.application.domain.payments.payment_details.rate_details import RateDetails

@dataclass
class DccProposal:
    base_amount: Optional[AmountOfMoney] = None
    disclaimer_display: Optional[str] = None
    disclaimer_receipt: Optional[str] = None
    rate: Optional[RateDetails] = None
    target_amount: Optional[AmountOfMoney] = None